feat(TreeView): warning for removal of selectable styling modifier

adamviktora opened the pull request:
Closes #601

wise-king-sullyman commented on packages/eslint-plugin-pf-codemods/src/rules/v6/treeViewWarnSelectableStylingModifierRemoved/treeView-warn-selectable-styling-modifier-removed.ts:
```diff
@@ -0,0 +1,32 @@
+    const treeViewImport = imports.find(
+      (specifier) => specifier.imported.name === "TreeView"
+    );
+
```
Could we add some logic so that it only throws this warning in the event that `pf-m-selectable` _would've_ been applied? 

For easy reference that logic was `children && (isSelectable || hasCheckbox)`

adamviktora replied:
After doing some investigation, we are not able to check it and we will have to throw a warning every time.

The reason is simple: the `children` and `hasCheckbox` props depend on the individual TreeViewDataItem objects from `data: TreeViewDataItem[]` prop of the TreeViewProps.

And since the TreeView may be used like this:
```
import {options} from "somewhere";
<TreeView data={options} ...
```
we are not able to trace it and check the condition.

wise-king-sullyman commented on packages/eslint-plugin-pf-codemods/src/rules/v6/treeViewWarnSelectableStylingModifierRemoved/treeView-warn-selectable-styling-modifier-removed.ts:
```diff
@@ -0,0 +1,32 @@
+              context.report({
+                node,
+                message:
+                  "Selectable styling attribute (`pf-m-selectable`) has been removed in TreeView component.",
```
Also could we just add a bit to the message here telling consumers what they might have to do because of this change/how it might impact them?

adamviktora replied:
Do you know what the message should say? Even after checking your React PR https://github.com/patternfly/patternfly-react/pull/10101 and the original issue I have no clue

wise-king-sullyman replied:
I would just let them know that they will need to update any selectors in their code or tests that depends on that classname being there, along with any snapshot tests that include the class.

wise-king-sullyman reviewed (changes requested):
Looks solid, two small comments

adamviktora commented:
I started addressing your comments and then I realized I made a mistake in the component name. The breaking change was on a `TreeViewListItem`, not `TreeView`.

TreeViewListItem is not directly exported from `@patternfly/react-core`, we only use this component internally. Although someone may import it with `import { TreeViewListItem } from '@patternfly/react-core/dist/esm/components/TreeView/TreeViewListItem';`

Do we still need a codemod just in case someone can import it?

wise-king-sullyman commented:
It's still pretty much on TreeView, just indirectly.

I think a codemod would still be good though since the list item class changes will still potentially impact TreeView users. I would just worry about it in that case though and not worry about someone importing it directly like that.

adamviktora commented:
Oh, I see, you are right. I was way too stuck in the "fixer" mindset from all the other rules 😃 where we only do changes if the exact component is imported.

wise-king-sullyman reviewed (approved):
🚀
